pkg/oc/cli/startbuild/startbuild.go

```diff
@@ -289,24 +288,24 @@ func (o *StartBuildOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, cmd
 			return fmt.Errorf("the provided Build %q was not created from a BuildConfig and cannot have webhooks", name)
 		}
 		if len(ref.Namespace) > 0 {
-			namespace = ref.Namespace
+			o.Namespace = ref.Namespace
 		}
 		name = ref.Name
 	}
 
-	o.Namespace = namespace
 	o.Name = name
 
-	o.BuildLogClient = buildclientinternalmanual.NewBuildLogClient(o.BuildClient.RESTClient(), o.Namespace)
+	o.BuildLogClient = buildclientmanual.NewBuildLogClient(o.BuildClient.RESTClient(), o.Namespace)
 
 	// Handle environment variables
 	cmdutil.WarnAboutCommaSeparation(o.ErrOut, o.Env, "--env")
+	// TODO: not sure what to do here, duplicate this helper to return v1 EnvVars?
 	env, _, err := utilenv.ParseEnv(o.Env, o.In)
```

Comment:
@deads2k @soltysh

Reply:
Do in-place conversion using `kcmdutil.AsDefaultVersionedOrOriginal` for now and leave there a TODO. In a followup to this and #20400 you'll sweep those at once.